### PR TITLE
Adds listener on "linkto" field

### PR DESCRIPTION
**Pull request by goldenapples**
Adds a callback function on the "linkto" field which shows or hides the
"custom URL" fields depending on whether the "Linkto" field is set to
"custom".

### CODE REVIEW

File: image-shortcake.php [hunk 1]
```diff
@@ -73,6 +74,32 @@ private function setup_filters() {
 
 
 	/**
+	 * Enqueue scripts and styles for editor admin area
+	 *
+	 * Defines the callback function which is run on chanes to any of the
+	 * shortcode attributes through the UI.
+	 */
+	public function enqueue_assets() {
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_shortcode_callback_js' ) );
```
Comment (danielbachhuber) [inline image-shortcake.php]: This should probably be run on `shortcode_ui_enqueue_scripts`

**Comment by goldenapples** [comment]
@danielbachhuber can you review and merge this?

File: image-shortcake.php [hunk 2]
```diff
@@ -73,6 +74,28 @@ private function setup_filters() {
...
+	}
+
+
+	/**
+	 * Enqueues the shortcode function on edit page
+	 *
+	 *
+	 * @action admin_enqueue_scripts
+	 */
+	public function enqueue_shortcode_js() {
```
Comment (danielbachhuber) [inline image-shortcake.php]: Can we prefix this with `action_` ?

File: assets/js/image-shortcake-admin.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+				customLinkField.$el.show()
+			} else {
+				customLinkField.$el.val('').hide();
+			}
+		}
+	}
+
+}
+
+wp.shortcake.hooks.addAction( 'img.linkto', ImageShortcake.listeners.linkto );
```
Comment (danielbachhuber) [inline assets/js/image-shortcake-admin.js]: What happens if we're using Shortcake v0.3.0 or earlier?

**Comment by danielbachhuber** [comment]
Couple questions.

**Comment by goldenapples** [comment]
Thanks. Fixed those issues. Want to take another look?

**Pull request merged**